spec/02-integration/03-db/02-db_core_entities_spec.lua

```diff
@@ -487,6 +487,7 @@ for _, strategy in helpers.each_strategy() do
             regex_priority  = 0,
             preserve_host   = false,
             strip_path      = true,
+            tags            = ngx.null,
```

Comment:
Should we return an empty set rather than `null`?

Reply:
This will follow the same pattern as for example `paths` for `routes`, which if empty will be set to `ngx.null`

Reply:
Those are different though: adding a path constricts the behaviour: it says "only consider this route when one of these array elements match" -> an empty path array would be effectively turning a route off.
Tags are different: an empty list of tags is the same as no tags, and so IMO should *always* be an array.
In practice this lets us remove lots of type checks, both inside of Kong, and in users of Kong.

Reply:
@james-callahan while I agree in principle that we should have been using empty sets rather than nulls everywhere (and I even disagree about your sub-point that paths are rightfully null, one could easily pick a semantics where empty set means no constraint added to paths), but this battle was [fought in 2017 and lost](https://github.com/Kong/kong/commit/cd748a3c9c14865c48de88e8b99f139fc6002419) (Ctrl+F "billion"), and now null is the default in all aggregate empty-accepting fields.

So, in short, @fffonion's design is fine, as it is consistent with the behavior of other attributes.